app/javascript/src/dashboard.js

```diff
@@ -116,24 +71,141 @@ $('document').ready(() => {
     }
   )
 
+  const handleAjaxError = e => {
+    if (e.status === 401) {
+      location.reload();
+    } else {
+      console.log(e);
+      if (e.responseJSON && e.responseJSON.error) {
+        alert(e.responseJSON.error);
+      } else if (e.response.statusText) {
+        alert(`Unable to complete your request: ${e.response.statusText}`);
+      } else {
+        alert("Unable to complete your request.");
+      }
+    }
+  };
+
   // Enable all data tables on dashboard but only filter on volunteers table
+  const editSupervisorPath = id => `/supervisors/${id}/edit`;
+  const editVolunteerPath = id => `/volunteers/${id}/edit`;
+  const casaCasePath = id => `/casa_cases/${id}`;
   var volunteersTable = $('table#volunteers').DataTable({
     autoWidth: false,
     stateSave: false,
-    columnDefs: [
+    columns: [
       {
-        targets: [1],
+        name: "display_name",
+        render: (data, type, row, meta) => {
+          return `
+            <a href="${editVolunteerPath(row.id)}">
+              ${row.display_name || row.email}
+              ${row.made_contact_with_all_cases_in_days === 'false'
+                ? `🕐 <i class="fa fa-question-circle" aria-hidden="true" data-toggle="tooltip" title="Has at least one case with no contact in 14 days"></i>`
+                : ''
+              }
+            </a>
+          `;
+        }
+      },
+      {
+        name: "email",
+        render: (data, type, row, meta) => row.email,
         visible: false
       },
       {
-        targets: [6],
+        className: "supervisor-column",
+        name: "supervisor_name",
+        render: (data, type, row, meta) => {
+          if (!row.supervisor.id) return "";
+
+          return `
+            <a href="${editSupervisorPath(row.supervisor.id)}">
+              ${row.supervisor.name}
+            </a>
+          `;
+        }
+      },
+      {
+        name: "active",
+        render: (data, type, row, meta) => row.active === "true" ? "Active" : "Inactive",
+        searchable: false,
+      },
+      {
+        name: "has_transition_aged_youth_cases",
+        render: (data, type, row, meta) => row.has_transition_aged_youth_cases === "true" ? "Yes 🐛🦋" : "No",
+        searchable: false
+      },
+      {
+        name: "casa_cases",
+        render: (data, type, row, meta) => {
+          const links = row.casa_cases.map(casaCase => {
+            return `<a href="${casaCasePath(casaCase.id)}">${casaCase.case_number}</a>`;
+          });
+
+          return links.join(", ");
+        },
+        orderable: false
+      },
+      {
+        name: "most_recent_contact_occurred_at",
+        render: (data, type, row, meta) => {
+          if (!row.most_recent_contact.case_id) return "None ❌";
```

Comment:
Please always use brackets with if statements in javascript. Right now the js linter is not enabled in CI but hopefully it will catch this when we turn it on.